clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -2000,6 +2003,58 @@ protected void processCommonConsumerAction(
     }
   }
 
+  /**
+   * This function checks, for a regular user store, if its persisted offset is greater than the end offset of its
+   * corresponding Kafka topic (indicating an offset rewind event, thus a potential data loss in Kafka) and increases
+   * related sensor counter values.
+   */
+  private void reportUserStoreOffsetRewindMetrics(PartitionConsumptionState pcs) {
+    if (!isUserSystemStore()) {
```

Comment:
Why are we excluding the user system stores? Is there a technical challenge? User system stores are still important and should be treated as first class citizens.

Reply:
This is a good catch! It should be applicable to all stores as you reasoned.